src/psyclone/psyir/nodes/container.py

```diff
@@ -55,11 +55,32 @@ class Container(Node):
     :type parent: :py:class:`psyclone.psyir.nodes.Node`
 
     '''
+    # Class attributes of this node textual representations
+    _children_valid_format = "*[Container | KernelSchedule]"
```

Comment:
I presume * means 0 or more? If so, I would expect the * to be at the end if we follow the usual regex format. Could you also add an explanation of the use of children_valid_format and the actual format used in the developers guide. While I am asking could you please add a section explaining what _validate_child does, what its requirements are and how it is implemented (i.e. the subclassing of list). Some of the info is already written in the PR. If you could also mention the need to add _colour_key and why as well that would be great.

Reply:
I moved * and + to the end and wrote a new developer_guide section.